tests/models/test_momentum_balance.py

```diff
@@ -289,3 +293,396 @@ def test_lithostatic(dim: int):
     # The stress at the bottom of the domain should be equal to the weight of the
     # column.
     assert np.allclose(bottom_traction, -rho * g * sd.cell_volumes.sum())
+
+
+"""Tests for elastoplastic split of fracture deformation below."""
+
+
+class ElastoplasticModel2d(
+    SquareDomainOrthogonalFractures,
+    pp.model_boundary_conditions.BoundaryConditionsMechanicsDirNorthSouth,
+    pp.models.momentum_balance.MomentumBalance,
+):
+
+    pass
+
+
+# We set a high shear modulus to make the domain stiff (especially for shear), and a
+# low tangential fracture stiffness to make the fracture weak. This will make the
+# fracture deform significantly more than the surrounding domain. Still, the domain
+# will deform slightly, requiring a tolerance when comparing the results.
+solid_vals_elastoplastic = {
+    "fracture_tangential_stiffness": 1e-5,
+    "shear_modulus": 1e6,
+    "lame_lambda": 1e2,
+}
+
+
+def verify_elastoplastic_deformation(
+    setup: pp.models.MomentumBalance,
+    u_e_expected: list[pp.number],
+    u_p_expected: list[pp.number],
+    u_top_expected: list[pp.number],
+    traction_expected: list[pp.number],
+    tols: list[float],
+    compare_means: bool = False,
+):
+    """Verify the results of a simulation.
+
+    The function will raise an assertion error if the results do not match the expected
+    values within the given tolerances.
+
+    Parameters:
+        setup: The model setup.
+        u_e_expected: ``len=nd``
+
+            Expected values of the elastic displacement jump in the x and y directions.
+        u_p_expected: ``len=nd``
+
+            Expected values of the plastic displacement jump in the x and y directions.
+        u_top_expected: ``len=nd``
+
+            Expected value of displacement in the cells above the fracture. nan values
+            are ignored, i.e., the test will pass if all computed values match the
+            corresponding non-nan expected values.
```

Comment:
Are nans used as a mask? Is there a reason masking is needed?

Reply:
Yes. If I don't really know what to expect or the values differ significantly between cells, I figured this was more explicit than just putting a huge tolerance.